spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -423,13 +454,42 @@ def show_context_menu(self, qpoint):
             point = self.figcanvas.mapToGlobal(qpoint)
             self.sig_context_menu_requested.emit(point)
 
+    def set_current_thumbnail(self, thumbnail):
+        """Set the current thumbnail displayed in the viewer."""
+        self.current_thumbnail = thumbnail
+
     def load_figure(self, fig, fmt):
         """Set a new figure in the figure canvas."""
+        self.auto_fit_plotting = self.current_thumbnail.auto_fit
+
+        # Let scale_image compute the scale factor for the thumbnail if it
+        # hasn't one yet.
+        if self.current_thumbnail.scalefactor is not None:
+            self.scalefactor = self.current_thumbnail.scalefactor
+
         self.figcanvas.load_figure(fig, fmt)
         self.sig_figure_loaded.emit()
         self.scale_image()
         self.figcanvas.repaint()
 
+        # Save the computed scale factor by scale_image in the thumbnail
+        if self.current_thumbnail.scalefactor is None:
+            self.current_thumbnail.scalefactor = self.scalefactor
+
+        # Restore scrollbar values
+        QTimer.singleShot(
+            20,
+            lambda: self.verticalScrollBar().setValue(
+                self.current_thumbnail.vscrollbar_value
+            ),
+        )
+        QTimer.singleShot(
+            20,
+            lambda: self.horizontalScrollBar().setValue(
+                self.current_thumbnail.hscrollbar_value
+            ),
+        )
```

Comment:
Maybe an explanation of why these timers need to be set here could be worthy

Reply:
Done at the top of these timers.